kolibri/core/api.py

```diff
@@ -39,14 +40,17 @@ def validate_token(self, request):
         PORTAL_URL = conf.OPTIONS["Urls"]["DATA_PORTAL_SYNCING_BASE_URL"]
         try:
             # token is in query params
-            response = requests.get(
-                join_url(
-                    PORTAL_URL, "portal/api/public/v1/registerfacility/validate_token"
-                ),
+            client = NetworkClient(PORTAL_URL)
+            response = client.get(
+                "portal/api/public/v1/registerfacility/validate_token",  # no need to join it, NetworkClient does it
                 params=request.query_params,
             )
-        except requests.exceptions.ConnectionError:
+        # as NetworkLocationResponseFailure is a subclass of requests.exceptions.ConnectionError
+        except errors.NetworkLocationConnectionFailure:
             return Response({"status": "offline"}, status=HTTP_503_SERVICE_UNAVAILABLE)
+        except errors.NetworkLocationResponseFailure as e:
```

Comment:
Initially i thought NetworkClient is not working properly as for this [endpoint](https://kolibridataportal.learningequality.org/portal/api/public/v1/registerfacility/validate_token/) i was getting a 404, but with requests it was going smoothly for the same endpoint. Turns out there were no exception handling for 404(directly using requests).
I think [endpoint](https://kolibridataportal.learningequality.org/portal/api/public/v1/registerfacility/validate_token/) should  return a 400 is the token is invalid(or NONE) instead of a 404???

Reply:
Looks like you could just set:
`response = e.response` here and let the code continue?